### ISSUE TITLE
Tags appear to be evaluated with OR instead of AND

### ISSUE THREAD

**Issue by csandfeld** [RamblingCookieMonster/PSDeploy#86]
Based on the description in issue #3 (and I think I saw it elsewhere also, but cannot find that right now) I expected only deployments tagged with *all* specified tags, to be invoked when I call invoke-psdeploy with multiple tags. But my tests show that the opposite is the case. Apparently all deployments that have any of the specified tags are invoked (see attached gif)

![2017-06-23_17-48-45](https://user-images.githubusercontent.com/2597673/27489838-622d14de-583c-11e7-8993-0193c8f23854.gif)

**Comment by RamblingCookieMonster** [comment]
Yeah, both the parameter docs and that are wrong, it currently uses the [equivalent of or](https://github.com/RamblingCookieMonster/PSDeploy/blob/master/PSDeploy/Private/Get-TaggedDeployment.ps1#L13)

Guessing we should:

* Update docs to indicate it's an `or`
* Add an option (everywhere a user can specify tags) to allow using `and`... e.g. `-TagOperator` with validateset, or an appropriately named switch param

Would prefer that to changing the default behavior, guessing that might break existing workflows for some folks

Thanks for pointing this out!

**Comment by csandfeld** [comment]
I agree - changing the current default behavior would come with a high risk of messing it up for existing deployments depending on tags.

A `-TagOperator` parameter or something similar would be nice.

**Comment by Line40** [comment]
Hi there, I would like to help with implementing this feature, in fact I already did implement a `-TagsAnd` switch for `Invoke-PSDeploy` that causes the Tags to be And'ed. However I don't particularly like the paramter name I chose, and maybe the PSDeploy way to do it would rather be something else. So i'd be happy to implement a different approach if there is one. 
I'm also open to implemting `-TagOperator` suggested by @csandfeld , but since, at least from how I understood what -TagOperator should be like, the only viable options would be `-TagOperator And` and `-TagOperator Or` I'm not sure if thats not overkill?
